### ISSUE TITLE
issues with `.env` with spaces and then the lack of a trailing newline

### ISSUE THREAD

**Issue by paperclover** [oven-sh/bun#411]
env files sometimes chop off the last character off when there is a space in the value, but not followed by a newline. here are my findings. pay attention to the trailing new line.

---

case 1: the bug
```
VARNAME=A B
```
VARNAME is `"A "`

---

case 2: a new line fixing the bug
```
VARNAME=A B

```
VARNAME is `"A B"`

---

case 3 and 4: bug not present without a space in the value
```
VARNAME=ABC
```
```
VARNAME=ABC

```
VARNAME is `"ABC"` in both cases

**Comment by Electroid** [comment]
Looks like this has been fixed as of `0.2.2`, tested on MacOS Silicon.

```txt
HELLO=A  B
```

```js
// index.ts
export default {
  fetch(request: Request) {
    return new Response(`'${process.env.HELLO}'`);
  }
}
```

```txt
> GET / HTTP/1.1
> Host: 0.0.0.0:3000
> User-Agent: curl/7.79.1
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 HM
< content-type: text/plain;charset=utf-8
< Date: Wed, 02 Nov 2022 22:04:42 GMT
< Content-Length: 10
< 
* Connection #0 to host 0.0.0.0 left intact
'A  B'
```

If you are able to reproduce this again, please comment and we can re-open the issue.

**Comment by paperclover** [comment]
can reproduce on Linux 6.0.7-arch1-1 on bun canary as of today.

can you attempt this again using the following command `echo -n 'VARNAME=A B' > .env`, to ensure the trailing newline does not exist, as most tools will ensure a trailing newline

![image](https://user-images.githubusercontent.com/24465214/200219084-c5b73a96-442a-45ee-ba27-93141ea31784.png)
